RecentActivity/src/org/sleuthkit/autopsy/recentactivity/ChromeCacheExtractor.java

```diff
@@ -0,0 +1,1265 @@
+        } 
+    }
+
+    ChromeCacheExtractor(Content dataSource, IngestJobContext context ) {
+        moduleName = NbBundle.getMessage(ChromeCacheExtractor.class, "ChromeCacheExtractor.moduleName");           
+        this.dataSource = dataSource;
+        this.context = context;
+    }
+    
+    
+    /**
+     * Initializes Chrome cache extractor module.
+     * 
+     * @throws IngestModuleException 
+     */
+    void moduleInit() throws IngestModuleException {
+        
+        try {
+            currentCase = Case.getCurrentCaseThrows();
+            fileManager = currentCase.getServices().getFileManager();
+             
+            // Create an output folder to save any derived files
+            absOutputFolderName = RAImageIngestModule.getRAOutputPath(currentCase, moduleName);
+            relOutputFolderName = RAImageIngestModule.getRelModuleOutputPath() + File.separator + moduleName;
+            File dir = new File(absOutputFolderName);
+            if (dir.exists() == false) {
+                dir.mkdirs();
+            }
+        } catch (NoCurrentCaseException ex) {
+            String msg = "Failed to get current case.";
+            throw new IngestModuleException(msg, ex);
+        } 
+    }
+    
+    /**
+     * Initializes the module to extract cache from a specific folder.
+     * 
+     * @param cachePath - path where cache files are found 
+     * 
+     * @throws org.sleuthkit.autopsy.ingest.IngestModule.IngestModuleException 
+     */
+    void subInit(String cachePath) throws IngestModuleException {
+        
+        filesTable.clear();
+        
+        String cacheAbsOutputFolderName = this.getAbsOutputFolderName() + cachePath;
+        File outDir = new File(cacheAbsOutputFolderName);
+        if (outDir.exists() == false) {
+            outDir.mkdirs();
+        }
+        
+        String cacheTempPath = RAImageIngestModule.getRATempPath(currentCase, moduleName) + cachePath;
+        File tempDir = new File(cacheTempPath);
+        if (tempDir.exists() == false) {
+            tempDir.mkdirs();
+        }
+    }
+    
+    /**
+     * Cleans up after the module is done.
+     * 
+     * Removes any temp copies of cache files created during extraction.
+     * 
+     */
+    void cleanup () {
+        
+        for (Entry<String, CacheFileCopy> entry : this.filesTable.entrySet()) {
+            String tempFilePathname = RAImageIngestModule.getRATempPath(currentCase, moduleName) + File.separator + entry.getKey(); 
+            try {
+                entry.getValue().getFileCopy().getChannel().close();
+                entry.getValue().getFileCopy().close();
+               
+                File tmpFile = new File(tempFilePathname);
+                if (!tmpFile.delete()) {
+                    tmpFile.deleteOnExit();
+                }
+            } catch (IOException ex) {
+                logger.log(Level.SEVERE, String.format("Failed to delete cache file copy %s", tempFilePathname), ex);
```

Comment:
If we are unable to delete a temp file is that a SEVERE error?  Maybe this should be information or a warning?

Reply:
I would call this a SEVERE error. Here is the way I think about the error levels:

INFO - Something normal/expected happened and we are logging it to records the flow of processing, establishing context for other messages. For example, the ingest job started and ingest job finished messages. 
WARNING - An error occurred but it is a result of the input, not a problem with the system. For example, it appears that a file is corrupted. Or library code experienced an error processing a file.
SEVERE - Something is wrong with the system or the code. For example, a column is missing from the case database or a case database query fails, i.e., many TskCoreExceptions. Or the RegRipper executable laid down by the installer is missing.

In this case, I would say that an inability to delete a temp file that we wrote says something is wrong with the system.

Reply:
LOL, I would have made it a Warning, were I not aware of Richard's sentiments on this subject, I've made that mistake a few times.

That said, I do agree, not being able to delete a temp file shouldn't  be a  SEVERE error, its does not have the same import and implications on working of Autopsy, as does not being able to find RegRipper, or being unable to open a database, or not finding an expected  column in a table.  

I often wish we had additional error levels like MAJOR and MINOR , in addition to SEVERE.  SEVERE, IMHO, should be reserved for "cannot proceed further, going to shut down now...".  This error would be a MINOR,

Reply:
Please hold that thought, we will continue to work towards a standard offline. Please call it whichever way you two want here.